src/renderer/features/currency/CurrencyForm/model/currency-form.ts

```diff
@@ -32,17 +34,19 @@ const $unpopularFiatCurrencies = currencyModel.$currencyConfig.map((config) => {
   return config.filter((c) => c.category === 'fiat' && !c.popular);
 });
 
-sample({
-  clock: priceProviderModel.$fiatFlag,
-  filter: (fiatFlag): fiatFlag is boolean => fiatFlag !== null,
-  target: $currencyForm.fields.fiatFlag.set,
-});
+type Params = {
+  fiatFlag: boolean | null;
+  currency: CurrencyItem | null;
+};
 
 sample({
-  clock: currencyModel.$activeCurrency,
-  filter: (currency): currency is CurrencyItem => currency !== null,
-  fn: (currency) => currency!.id,
-  target: $currencyForm.fields.currency.set,
+  clock: formInitiated,
```

Comment:
What if `priceProviderModel.$fiatFlag` & `currencyModel.$activeCurrency` are still "loading"?
How can I set the form state when those stores will have finished?